src/MultiFormats/MultiHash.cs

```diff
@@ -0,0 +1,587 @@
+        /// </returns>
+        /// <seealso cref="ToBase58"/>
+        public override string ToString() { return ToBase58(); }
+
+        /// <summary>
+        ///   Returns the <see cref="Base58"/> encoding of the <see cref="MultiHash"/>.
+        /// </summary>
+        /// <returns>
+        ///   The <see cref="Base58"/> representation of the <see cref="MultiHash"/>.
+        /// </returns>
+        public string ToBase58()
+        {
+            if (_b58String != null) return _b58String;
+
+            using (var ms = new MemoryStream())
+            {
+                Write(ms);
+                _b58String = ms.ToArray().ToBase58();
+                return _b58String;
+            }
+        }
+
+        /// <summary>
+        ///   Returns the <see cref="Base32"/> encoding of the <see cref="MultiHash"/>.
+        /// </summary>
+        /// <returns>
+        ///   The <see cref="Base32"/> representation of the <see cref="MultiHash"/>.
+        /// </returns>
+        public string ToBase32() { return ToArray().ToBase32(); }
+
+        /// <summary>
+        ///   Returns the IPFS binary representation as a byte array.
+        /// </summary>
+        /// <returns>
+        ///   A byte array.
+        /// </returns>
+        /// <remarks>
+        ///   The binary representation is a sequence of <see cref="MultiHash"/>.
+        /// </remarks>
+        public byte[] ToArray()
+        {
+            using (var ms = new MemoryStream())
+            {
+                Write(ms);
+                return ms.ToArray();
+            }
+        }
+
+        /// <summary>
+        ///   Determines if the data matches the hash.
+        /// </summary>
+        /// <param name="data">
+        ///   The data to check.
+        /// </param>
+        /// <returns>
+        ///   <b>true</b> if the data matches the <see cref="MultiHash"/>; otherwise, <b>false</b>.
+        /// </returns>
+        /// <remarks>
+        ///   <b>Matches</b> is used to ensure data integrity.
+        /// </remarks>
+        public bool Matches(byte[] data)
+        {
+            var digest = Algorithm.Hasher().ComputeHash(data);
+            for (var i = digest.Length - 1; 0 <= i; --i)
+                if (digest[i] != Digest[i])
+                    return false;
+
+            return true;
+        }
+
+        /// <summary>
+        ///   Determines if the stream data matches the hash.
+        /// </summary>
+        /// <param name="data">
+        ///   The <see cref="Stream"/> containing the data to check.
+        /// </param>
+        /// <returns>
+        ///   <b>true</b> if the data matches the <see cref="MultiHash"/>; otherwise, <b>false</b>.
+        /// </returns>
+        /// <remarks>
+        ///   <b>Matches</b> is used to ensure data integrity.
+        /// </remarks>
+        public bool Matches(Stream data)
+        {
+            var digest = Algorithm.Hasher().ComputeHash(data);
+            for (var i = digest.Length - 1; 0 <= i; --i)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'for' block.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)